src/time.cpp

```diff
@@ -87,20 +87,27 @@ namespace g3 {
 
    int timespec_get(struct timespec* ts/*, int base*/) {
 #ifdef __MACH__
-      // std::timespec_get or posix clock_gettime)(...) are not
-      // implemented on OSX
-      // @return value of base if successful, else zero
-      struct timeval now = {};
-      int rv = gettimeofday(&now, nullptr);
-      if (-1 == rv) {
-         return rv;
-      }
-      // error mode. just return sec, microsecond
-      ts->tv_sec  = now.tv_sec;
-      ts->tv_nsec = now.tv_usec * 1000;
-      return 0;
+	   // std::timespec_get or posix clock_gettime)(...) are not
+	   // implemented on OSX
+	   // @return value of base if successful, else zero
+	   struct timeval now = {};
+	   int rv = gettimeofday(&now, nullptr);
+	   if (-1 == rv) {
+		   return rv;
+	   }
+	   // error mode. just return sec, microsecond
+	   ts->tv_sec = now.tv_sec;
+	   ts->tv_nsec = now.tv_usec * 1000;
+	   return 0;
 #elif(defined(WIN32) || defined(_WIN32) || defined(__WIN32__))
-	   return timespec_get(ts, TIME_UTC);
+	   static auto os =
```

Comment:
the time since epoch for the steady_clock is not necessarily really the time since 1970. It usually is the time since program start. 
Thus, here I calculate the offset between this (different) starting point and the real start of the epoch as reported by the system clock during the first call with the precision of the system clock. That means that in absolute terms time stamps will later have system clock accuracy but relative times will have the precision of the high resolution clock.
For my previous logging solutions had used only the steady clock and had the sink side calculate the absolute time in a similar way (by adding the offset). In fact it is an option to move some of this math to the logging thread (by allowing the message to be modified before passing it to the sinks). This would save time in the execution thread. Similarly splitPath() could be moved there, leaving _file empty in saveMessage and only populating _file_path (in case you accept #138)